app/src/main/java/ch/hikemate/app/ui/components/ElevationGraph.kt

```diff
@@ -23,34 +32,68 @@ fun ElevationGraph(
-
-    val widthStep = width / elevationDataAveraged.size
-
-    for (i in elevationDataAveraged.indices.take(elevationDataAveraged.size - 1)) {
-      // Scale the points to fit the whole width
-      val x = i * widthStep
-      // Scale the points to fit the whole height, but keeping the scale.
-      // The height is inverted because the origin of the canvas is at the top left corner
-      val y = (height - (elevationDataAveraged[i] - minElevation) / elevationStep).toFloat()
-      val nextX = (i + 1) * widthStep
-      val nextY = (height - (elevationDataAveraged[i + 1] - minElevation) / elevationStep).toFloat()
-      drawLine(color, Offset(x, y), Offset(nextX, nextY), strokeWidth = strokeWidth)
-    }
+  if (elevationData.isEmpty()) {
+    return
   }
+
+  // Averaging the elevation data to reduce the number of points
+  val averageSize = remember { ceil(elevationData.size.toDouble() / maxNumberOfPoints).toInt() }
+  val elevationDataAveraged = remember {
+    elevationData.chunked(averageSize) { chunk -> chunk.average() }
+  }
+
+  var height by remember { mutableFloatStateOf(0.0f) }
+  var width by remember { mutableFloatStateOf(0.0f) }
+
+  // Bunch of variables to scale the graph in height
+  val maxElevation = remember(elevationDataAveraged) { elevationDataAveraged.maxOrNull() ?: 0.0 }
+  val minElevation = remember(elevationDataAveraged) { elevationDataAveraged.minOrNull() ?: 0.0 }
+  val elevationRange = maxElevation - minElevation
+  val elevationStep = elevationRange / height
+
+  val widthStep = width / (elevationDataAveraged.size - 1)
+
+  // Build (x, y) tuples to draw the path
+  // Scale the points width to fit the whole width
+  // Scale the points height to fit the whole height, but keeping the scale.
+  // The height is inverted because the origin of the canvas is at the top left corner
+  val scaledData =
+      remember(height, width) {
+        elevationDataAveraged
+            .mapIndexed { index, elevation ->
+              Pair(
+                  index * widthStep,
+                  (height - (elevation - minElevation) / elevationStep).toFloat())
+            }
+            .map {
+              // We clamp the points to ensure they are not outside the canvas
+              Pair(it.first, MathUtils.clamp(it.second, strokeWidth, height - strokeWidth))
+            }
+      }
+
+  val path = Path()
+  path.moveTo(scaledData.first().first, scaledData.first().second)
+
+  for ((x, y) in scaledData.drop(1)) {
+    path.lineTo(x, y)
+  }
+
+  val outlinePath = path.copy()
+  outlinePath.lineTo(width, height)
+  outlinePath.lineTo(0f, height)
+
+  Canvas(
+      modifier.onGloballyPositioned { coordinates ->
+        height = coordinates.size.height.toFloat()
+        width = coordinates.size.width.toFloat()
+      }) {
+        drawPath(path, strokeColor, style = Stroke(width = strokeWidth))
+        clipPath(outlinePath, clipOp = androidx.compose.ui.graphics.ClipOp.Intersect) {
```

Comment:
Tied to last remark, should change to this:
```suggestion
        clipPath(path, clipOp = androidx.compose.ui.graphics.ClipOp.Intersect) {
```